test/functional/tool_wallet.py

```diff
@@ -212,6 +212,92 @@ def test_salvage(self):
+        dump_data = 'BITCOIN_CORE_WALLET_DUMP,0\n' + dump_data[27:] # Change the version number
+        with open(bad_ver_wallet_dump, 'w+', encoding='utf8') as f:
+            f.write(dump_data)
+        self.assert_raises_tool_error('Error: Dumpfile version is not supported. This version of bitcoin-wallet only supports version 1 dumpfiles. Got dumpfile with version 0', '-wallet=load1', '-format=bdb', '-dumpfile={}'.format(bad_ver_wallet_dump), 'createfromdump')
+        bad_ver_wallet_dump = os.path.join(self.nodes[0].datadir, "wallet-bad_ver2.dump")
+        dump_data = 'BITCOIN_CORE_WALLET_DUMP,2\n' + dump_data[27:] # Change the version number
+        with open(bad_ver_wallet_dump, 'w+', encoding='utf8') as f:
+            f.write(dump_data)
+        self.assert_raises_tool_error('Error: Dumpfile version is not supported. This version of bitcoin-wallet only supports version 1 dumpfiles. Got dumpfile with version 2', '-wallet=load2', '-format=bdb', '-dumpfile={}'.format(bad_ver_wallet_dump), 'createfromdump')
+        bad_magic_wallet_dump = os.path.join(self.nodes[0].datadir, "wallet-bad_magic.dump")
+        dump_data = 'not_the_right_magic,1\n' + dump_data[27:]
+        with open(bad_magic_wallet_dump, 'w+', encoding='utf8') as f:
+            f.write(dump_data)
+        self.assert_raises_tool_error('Error: Dumpfile identifier record is incorrect. Got "not_the_right_magic", expected "BITCOIN_CORE_WALLET_DUMP".', '-wallet=load3', '-format=bdb', '-dumpfile={}'.format(bad_magic_wallet_dump), 'createfromdump')
+
+        self.log.info('Checking createfromdump handling of checksums')
+        bad_sum_wallet_dump = os.path.join(self.nodes[0].datadir, "wallet-bad_sum1.dump")
+        checksum = dump_data[-65:-1]
+        dump_data = orig_dump[:-65] + "1" * 64 + "\n"
+        with open(bad_sum_wallet_dump, 'w+', encoding='utf8') as f:
+            f.write(dump_data)
+        self.assert_raises_tool_error('Error: Dumpfile checksum does not match. Computed {}, expected {}'.format(checksum, "1" * 64), '-wallet=load4', '-format=bdb', '-dumpfile={}'.format(bad_sum_wallet_dump), 'createfromdump')
+        bad_sum_wallet_dump = os.path.join(self.nodes[0].datadir, "wallet-bad_sum2.dump")
+        dump_data = orig_dump[:-75]
+        with open(bad_sum_wallet_dump, 'w+', encoding='utf8') as f:
+            f.write(dump_data)
+        self.assert_raises_tool_error('Error: Missing checksum', '-wallet=load5', '-format=bdb', '-dumpfile={}'.format(bad_sum_wallet_dump), 'createfromdump')
+        bad_sum_wallet_dump = os.path.join(self.nodes[0].datadir, "wallet-bad_sum3.dump")
+        dump_data = orig_dump[:-65] + "2" * 10
+        with open(bad_sum_wallet_dump, 'w+', encoding='utf8') as f:
+            f.write(dump_data)
+        self.assert_raises_tool_error('Error: Dumpfile checksum does not match. Computed {}, expected {}{}'.format(checksum, "2" * 10, "0" * 54), '-wallet=load6', '-format=bdb', '-dumpfile={}'.format(bad_sum_wallet_dump), 'createfromdump')
```

Comment:
In commit "tests: Test bitcoin-wallet dump and createfromdump" (35442fa568583c0da2c687966e49b3742f403870)

Looking at the `CreateFromDump` implementation it seems like the wallet will be created but empty if there is a checksum error or other import error. It would be good if there was a check at this point to verify if the created wallets are loadable if they are supposed to be loadable or not loadable if they are not supposed be loadable.

Reply:
I've added a check that the file exists (or doesn't for failure cases).